.gitlab-ci.yml

```diff
@@ -67,6 +67,25 @@ caramel:flake:
   script:
     - flake8 caramel/ tests/
 
+caramel:mypy:
+  stage: test
+  when: always
+  image: ${PYTHON_IMAGE}
+  before_script:
+    - pip3 install mypy
```

Comment:
I compleatly missed that, thought that it was unofficial since mypy didn't find it on its own. Anyway now we have a fun error with mypy where it crashes on an internal error. 

``` caramel/models.py:246: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.910
Traceback (most recent call last):
  File "mypy/semanal.py", line 4872, in accept
  File "mypy/nodes.py", line 950, in accept
  File "mypy/semanal.py", line 1056, in visit_class_def
  File "mypy/semanal.py", line 1134, in analyze_class
  File "mypy/semanal.py", line 1143, in analyze_class_body_common
  File "mypy/semanal.py", line 1203, in apply_class_plugin_hooks
  File "/usr/local/lib/python3.8/dist-packages/SQLAlchemy-1.4.3-py3.8-linux-x86_64.egg/sqlalchemy/ext/mypy/plugin.py", line 173, in _base_cls_hook
    decl_class._scan_declarative_assignments_and_apply_types(ctx.cls, ctx.api)
  File "/usr/local/lib/python3.8/dist-packages/SQLAlchemy-1.4.3-py3.8-linux-x86_64.egg/sqlalchemy/ext/mypy/decl_class.py", line 123, in _scan_declarative_assignments_and_apply_types
    _scan_declarative_assignment_stmt(cls, api, stmt, cls_metadata)
  File "/usr/local/lib/python3.8/dist-packages/SQLAlchemy-1.4.3-py3.8-linux-x86_64.egg/sqlalchemy/ext/mypy/decl_class.py", line 377, in _scan_declarative_assignment_stmt
    type_id = names._type_id_for_callee(stmt.rvalue.callee)
  File "/usr/local/lib/python3.8/dist-packages/SQLAlchemy-1.4.3-py3.8-linux-x86_64.egg/sqlalchemy/ext/mypy/names.py", line 168, in _type_id_for_callee
    return _type_id_for_funcdef(callee.node)
  File "/usr/local/lib/python3.8/dist-packages/SQLAlchemy-1.4.3-py3.8-linux-x86_64.egg/sqlalchemy/ext/mypy/names.py", line 179, in _type_id_for_funcdef
    if hasattr(node.type.ret_type, "type"):
AttributeError: 'NoneType' object has no attribute 'ret_type'
caramel/models.py:246: : note: use --pdb to drop into pdb ```

I don't really understand it, something to do with the models.AccessLog class( the Columns ?)

Reply:
Outdated sqlalchemy install on my end(1.4.3), version 1.4.20 works.